sdk/keyvault/remove-test-resources-pre.ps1

```diff
@@ -0,0 +1,79 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+
+# IMPORTANT: Do not invoke this file directly. Please instead run eng/common/TestResources/Remove-TestResources.ps1 from the repository root.
+
+#Requires -Version 6.0
+#Requires -PSEdition Core
+
+# Use same parameter names as declared in eng/common/TestResources/Remove-TestResources.ps1 (assume validation therein).
+[CmdletBinding(DefaultParameterSetName = 'Default', SupportsShouldProcess = $true, ConfirmImpact = 'Medium')]
+param (
+    [Parameter()]
+    [string] $ResourceGroupName,
+
+    [Parameter()]
+    [string] $TenantId,
+
+    [Parameter()]
+    [string] $ProvisionerApplicationId,
+
+    [Parameter()]
+    [string] $ProvisionerApplicationSecret,
+
+    [Parameter()]
+    [string] $SubscriptionId,
+
+    # Captures any arguments from eng/New-TestResources.ps1 not declared here (no parameter errors).
+    [Parameter(ValueFromRemainingArguments = $true)]
+    $RemainingArguments
+)
+
+# By default stop for any error.
+if (!$PSBoundParameters.ContainsKey('ErrorAction')) {
+    $ErrorActionPreference = 'Stop'
+}
+
+function Log($Message) {
+    Write-Host ('{0} - {1}' -f [DateTime]::Now.ToLongTimeString(), $Message)
+}
+
+function PurgeKeyVault($Vault) {
+  Log "Deleting Key Vault named '$($Vault.VaultName)'"
+  Remove-AzKeyVault -Name "$($Vault.VaultName)" -ResourceGroupName "$($Vault.ResourceGroupName)" -Location $($Vault.Location) -Force
+  Log "Deleted."
+
+  Log "Purging Key Vault named '$($Vault.VaultName)'"
+  Remove-AzKeyVault -Name "$($Vault.VaultName)" -Location $($Vault.Location) -InRemovedState -Force
+
+  Log "'$($Vault.VaultName)' successfully deleted and purged."
+}
+
+function PurgeManagedHsm($ManagedHsm) {
+  Log "Deleting Managed HSM named '$($ManagedHsm.Name)'"
+  az keyvault delete --resource-group "$ResourceGroupName" --hsm-name "$($ManagedHsm.Name)"
+  Log "Deleted."
+
+  Log "Purging Managed HSM named '$($ManagedHsm.Name)'"
+  az keyvault purge --hsm-name "$($ManagedHsm.Name)"
```

Comment:
It's interesting that this does not require a location like Key Vault above. Just an observation.

Reply:
Yea for sure, I didn't find it in the docs so I assumed it's not needed. I think the other doesnt require a location either but I figured it's good to be specific!